### PR TITLE
Remove version information from app name

### PR DESCRIPTION
**Pull request by xhaggi**
If the war archive or exploded war folder contains version information, it should be removed from the app name.

* `my-app-1.0-SNAPSHOT` becomes `my-app`
* `my-app-1.0` becomes `my-app`

### CODE REVIEW

**Comment by cedric05** [comment]
refer https://github.com/adashen/vscode-tomcat/pull/244#issuecomment-503609476 for failure details

**Comment by nicolaslohrer** [comment]
This looks like it might solve an issue that I have with my setup. Any chance to get this merged?

**Comment by xhaggi** [comment]
@andxu mind taking a look at it?

File: src/Tomcat/TomcatController.ts [hunk 1]
```diff
@@ -356,11 +356,21 @@ export class TomcatController {
...
-        const defaultName: string = path.basename(webappPath, path.extname(webappPath));
-        let appName: string = defaultName;
+        const isWarFile: boolean = this.isWarFile(webappPath);
+
+        let appName: string = isWarFile
+            ? path.basename(webappPath, path.extname(webappPath))
+            : path.basename(webappPath);
+
+        // remove version information from app name if any
+        const versionIndex: number = appName.search(/\-[0-9]+\.[0-9]+/gi);
```
Comment (andxu) [inline src/Tomcat/TomcatController.ts:366]: the regex is not compatible with most versions like snapshot,release,beta,alpha, I suggest to also try to find the artifact at 
META-INF\maven\...\pom.xml inside the war file

Comment (xhaggi) [reply src/Tomcat/TomcatController.ts:366]: I'm not sure what you exactly mean. The regex finds the first version part `1.0` of `1.0-SNAPSHOT` or `1.0.RELEASE` etc. removes it and what comes after it.

Comment (andxu) [reply src/Tomcat/TomcatController.ts:366]: I think I known you meaning, you find the index of '1.0 ' and then substr to that index, then I think you'd better modify the regex to match the last occurrence of slash and number

**Pull request closed without merging**